tests/component/component.cpp

```diff
@@ -119,7 +119,7 @@ TEST(Component, addAndCountChildren)
 
     child3->addComponent(child4);
     parent->addComponent(child3);
-    EXPECT_EQ(size_t(3), parent->componentCount());
+    EXPECT_EQ(size_t(4), parent->componentCount());
```

Comment:
I would be curious to see what the resulting model looks like at this stage...

Reply:
```
<?xml version="1.0" encoding="UTF-8"?>
<model xmlns="http://www.cellml.org/cellml/2.0#">
  <component name="parent"/>
  <component name="child1"/>
  <component name="child2"/>
  <component name="child3"/>
  <component name="child4"/>
  <component name="child3"/>
  <component name="child4"/>
  <encapsulation>
    <component_ref component="parent">
      <component_ref component="child1"/>
      <component_ref component="child2"/>
      <component_ref component="child3">
        <component_ref component="child4"/>
      </component_ref>
      <component_ref component="child3">
        <component_ref component="child4"/>
      </component_ref>
    </component_ref>
  </encapsulation>
</model>
```

For your curiosity.  I don't think the test needs this though.

Reply:
Agreed, we don't. This aside, to allow for this kind of behaviour doesn't make sense to me, but you guys seem to be ok with it, so whatever.